flocker/acceptance/testtools.py

```diff
@@ -911,6 +911,13 @@ def detach_and_destroy_volume(volume):
                 d.addBoth(
                     lambda _: async_api.destroy_volume(volume.blockdevice_id)
                 )
+                # Consume failures and write them out. Failures might just
+                # indicate that the cluster beat us to deleting the volume
+                # which should not cause the test to fail, we only want the
+                # test to fail if list_volumes still returns a non-empty list.
+                # The construction of api_clean_state should prevent this from
+                # masking significant failures to clean up volumes.
+                d.addErrback(write_failure)
```

Comment:
:grey_question: Sounds correct. But does this also apply to non GCE tests? I remember you saying something about instantiating another copy of the GCE driver in the TestCase in order to do cleanup, but I don't think we do that for other drivers.